module/zfs/zio.c

```diff
@@ -145,6 +145,21 @@ static const int zio_buf_debug_limit = 16384;
 static const int zio_buf_debug_limit = 0;
 #endif
 
+typedef struct zio_stats {
+	kstat_named_t ziostat_gang_writes;
+	kstat_named_t ziostat_gang_multilevel;
+} zio_stats_t;
+
+static zio_stats_t zio_stats = {
+	{ "gang_writes",	KSTAT_DATA_UINT64 },
+	{ "gang_multilevel",	KSTAT_DATA_UINT64 },
+};
+
+#define	ZIOSTAT_BUMP(stat) \
+	atomic_inc_64(&zio_stats.stat.value.ui64);
```

Comment:
Global atomics aren't good.  wrsums would be better, especially if for reference you would also account a total number of allocations, special to normal fallbacks and something else you might think around that.